Dog resourses. For dog owners

elgaard opened the pull request:
Opening a pull request on MapComplete
=====================================

Hey! Thanks for opening a pull request on Mapcomplete! This probably means you want to add a new theme - if so, please
follow the checklist below. If this pull request is for some other issue, please ignore the template.

Adding your new theme
---------------------

Thanks for taking the time to create your theme and to add it to the main repository!

To making merging smooth, please make sure that each of the following conditions is met:

- [x] The codebase is GPL-licensed. By opening a pull request, the new theme will be GPL too
- [x] All images are included in the pull request and no images are loaded from an external service (e.g. Wikipedia)
- [x] The [guidelines on how to make your own theme](https://github.com/pietervdvn/MapComplete/blob/master/Docs/Making_Your_Own_Theme.md)
  are read and followed

pietervdvn commented:
Hi Elgaard,

Thanks for your pull request! The theme looks already pretty cool and I got it working without problems, so that is already pretty good!

I'm very strict when I open up a new theme, so there are a few issues to address - but I'm definitively planning on joining this.

# The formalities

Please check the checkmark above where you agree to the GPL-license to redistribute your theme.
The licenses for the artworks are already included, so that is pretty great!

# Language support

This is the first time that Danish is seen in my repository. Would you mind adding some translations to [MapComplete-core](https://hosted.weblate.org/projects/mapcomplete/core/da/) too?

# Presets?

It seems like no presets are defined; this means that contributors can not add new shops, restaurants or dog parks. Is this intentional?

# Better filtering

On a related note, the 'shops' and 'restaurants' will probably benefit from being reused. I have updated the [theme guidelines](https://github.com/pietervdvn/MapComplete/blob/develop/Docs/Making_Your_Own_Theme.md#what-is-a-good-theme) which offer a few best practices to have presets without introducing duplicates and with having filters in place.

pietervdvn commented:
Wooow, I didn't expect you to translate _everything_ ;) I would have been happy with just a few sentences... But I'm very happy with your thorough translation work!

elgaard commented:
Pieter Vander Vennet:
> Wooow, I didn't expect you to translate /everything/ ;) I would have been happy with 
> just a few sentences... But I'm very happy with your thorough translation work!

I though I better start with that. Because then I will get to see my own translations 
while working on the theme. (eating you own dog food :-)

I also checked the GPL and added some presets and more tag questions.

The theme now reuse the food and shops layers the same way you did with fries places.

The reason I do not just use the food layer is that I want the extra filtering 
because unfortunately there are very few restaurants and shops in Denmark that allow 
dogs. So when I look for one, I need to zoom out to all of Copenhagen if I am at home 
and plan to walk or use public transport, or zoom out to about half of Denmark if I 
am driving to Germany and want to eat on the way. And there are about 20000 shops, 
restaurants, etc.

Of course the fact that there are so few dog-friendly restaurant is also the reason I 
want to have restaurant on the theme. In E.g., Berlin most restaurants allows dogs, 
so it is less important to tag it, but in Denmark it is important to know every 
single restaurant that allow dogs.


-- 
Niels Elgaard Larsen

pietervdvn commented on assets/themes/dog/dog.json:
```diff
@@ -0,0 +1,427 @@
+        },
+        {
+          "id": "bin",
+          "mappings": [
```
I'm not very fond of this mixing of objects as it complicates matters.

What if we want to indicate that the bin is only for a certain kind of waste?
What if one wants to add a picture?

What if the waste basket is two meters away from the dog park? Should it be 'yes' or 'no'? What if there are many waste baskets?

Instead, include the `waste_basket` layer instead. (Bonus: it has a question on whether or not it has a dispenser for poo-bags)

pietervdvn commented:
> The theme now reuse the food and shops layers the same way you did with fries places.

Excellent!
The reason to include the _entire_ food layer at zoom level 19 is that it prevents duplicate entries from being added; so the way you did it is perfectly fine.

pietervdvn commented on assets/themes/dog/dog.json:
```diff
@@ -0,0 +1,427 @@
+            {
+              "if": "barrier=fence",
+              "then": {
+                "en": "fenced: Yes",
```
If the property is already known, this will be shown 'as is' to the user. It is better to use a full sentence, e.g. "This dogpark is fenced all around" instead of "fenced: yes". Same for "fenced: no"

pietervdvn commented on assets/themes/dog/dog.json:
```diff
@@ -0,0 +1,427 @@
+        {
+          "id": "dogarea",
+          "render": {
+            "en": "areal: {_surface:ha} ha",
```
Another small nitpick: this would better be prhased as `This dogpark is {_surface:ha} ha big` as it is friendlier for humans.

pietervdvn commented:
Allright, we're now at a point in this PR where we are nitpicking on details ;)

Apart from my remarks above, I'm not entirely happy yet with the symbol for vets: it contains too much detail for a small icon on a map. The cat/dog outlines become really small and unrecognizable. I propose to either use a _blue_ cross (as a blue cross is the symbol for a vet in Belgium - but that might be only here) or to ~~steal~~ reuse [this icon](https://upload.wikimedia.org/wikipedia/commons/8/87/Map_icons_by_Scott_de_Jonge_-_veterinary-care.svg) which the iD-editor uses too.

Apart from that: I'm really impressed with your theme!

elgaard commented:
On Wed, 27 Apr 2022 05:39:36 -0700
Pieter Vander Vennet ***@***.***> wrote:

> Allright, we're now at a point in this PR where we are nitpicking on
> details ;)
> 
> Apart from my remarks above, I'm not entirely happy yet with the
> symbol for vets: it contains too much detail for a small icon on a
> map.

I igree
> The cat/dog outlines become really small and unrecognizable. I
> propose to either use a _blue_ cross (as a blue cross is the symbol
> for a vet in Belgium - but that might be only here) 

I think so. Here it would refer to
https://en.wikipedia.org/wiki/International_Blue_Cross

But it seems to used at least in the UK too:
https://en.wikipedia.org/wiki/Blue_Cross_(animal_charity)

I would prefer something like 
https://commons.wikimedia.org/wiki/File:Veterinary_medicine_icon.jpg

except that is a jpg and a SA license


> or to ~~steal~~
> reuse [this
> icon](https://upload.wikimedia.org/wikipedia/commons/8/87/Map_icons_by_Scott_de_Jonge_-_veterinary-care.svg)
> which the iD-editor uses too.

I will use it or the blue cross if we cannot agree on anything better.
But I think the paw is too dominating on a theme that has other dog
symbols.

 
> Apart from that: I'm really impressed with your theme! 

Thank you 

/
-- 
Niels

elgaard commented:
On Wed, 27 Apr 2022 05:39:36 -0700
Pieter Vander Vennet ***@***.***> wrote:

> Allright, we're now at a point in this PR where we are nitpicking on
> details ;)

I believe that I have now addressed all your points

> Apart from my remarks above, I'm not entirely happy yet with the
> symbol for vets: it contains too much detail for a small icon on a
> map. The cat/dog outlines become really small and unrecognizable. I
> propose to either use a _blue_ cross (as a blue cross is the symbol
> for a vet in Belgium - but that might be only here) or to ~~steal~~
> reuse [this
> icon](https://upload.wikimedia.org/wikipedia/commons/8/87/Map_icons_by_Scott_de_Jonge_-_veterinary-care.svg)
> which the iD-editor uses too.
> 
> Apart from that: I'm really impressed with your theme! 
> 



-- 
Niels

pietervdvn commented:
Hi,

I merged in your theme! Thanks for contributing!

The next step is making some publicity for it - would you mind writing a diary entry about it?

The story about your puppy and the vet that was still open is really cute, it would be pretty nice to have it in there too :relaxed:
